JASP-Engine/JASP/R/mlClassificationBoosting.R

```diff
@@ -168,8 +168,12 @@ mlClassificationBoosting <- function(jaspResults, dataset, options, ...) {
     score <- predict(bfitAUC, newdata = test, n.trees = noOfTrees, type = "response")
     actual.class <- test[,.v(options[["target"]])] == lvls[i]
 
-    pred <- ROCR::prediction(score, actual.class)
-    auc[i] <- ROCR::performance(pred, "auc")@y.values[[1]]
+    if(length(levels(factor(actual.class))) == 2){
```

Comment:
Can we do something about these `for(i in 1:length(lvls)){` loops?
They do the same basic thing and share a lot of code. Which is exemplified by the fact you needed to copy-paste this fix 4 times.

(Also, above this it does ` typeData <- typeData[, -column]` twice)

Reply:
like?

Reply:
I made a pull request to your repo: https://github.com/koenderks/jasp-desktop/pull/11
This removes all the duplication. Should be checked though!